src/GridCal/Engine/Core/admittance_matrices.py

```diff
@@ -210,6 +210,26 @@ def compute_fast_decoupled_admittances(X, B, m, mf, mt, Cf, Ct):
     return B1.tocsc(), B2.tocsc()
 
 
+def add_shunt_fault_impedances(C_bus_elm, r, x):
```

Comment:
should this function not be in GeneratorData and BatteryData? There are simmilar ones to convert the power injections vector per device to vector per bus

Reply:
True, it is now improved in 865579a. I have added the functions to the generator and battery classes. In fact, the `add_shunt_fault_impedances` function is removed, it is not really needed with these modifications.